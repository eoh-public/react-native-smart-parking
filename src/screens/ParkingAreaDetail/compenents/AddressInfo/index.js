import React, { memo, useEffect } from 'react';
import { StyleSheet, View } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import moment from 'moment';
import { t } from 'i18n-js';

import Text from '../../../../commons/Text';
import { Colors, API } from '../../../../configs';
import { calcDistance } from '../../../../utils/Converter/distance';
import { formatMoney } from '../../../../utils/Utils';
import { axiosGet } from '../../../../utils/Apis/axios';
import { getCurrentLatLng } from '../../../../utils/CountryUtils';
import Routes from '../../../../utils/Route';
import { ParkingStatusBar, RowItem } from '../ParkingDetail';
import { ExpandView } from '../../../../commons';
import {
  SvgParking,
  SvgNavigate,
} from '../../../../../assets/images/SmartParking';
import { IconOutline } from '@ant-design/icons-react-native';
import { TESTID } from '../../../../configs/Constants';

const AddressInfo = memo(
  ({
    name,
    address,
    available_spots_count,
    parking_charges,
    free_time,
    status,
    distance: rawDistance,
    total_spot,
    preBook,
    spot_name,
    searchedLocation,
  }) => {
    const distance = calcDistance(rawDistance);
    const { navigate } = useNavigation();

    let start = '';
    let end = '';
    let freeFrom = '';
    let freeTo = '';
    let list_time = [];
    let time_now = '';
    let list_price = [];
    let price_now = '';
    let titleColor = Colors.Gray9;

    // TODO: detect now is in any timezone
    if (free_time) {
      freeFrom = moment(`2020-01-01T${free_time.free_from}`).format('LT');
      freeTo = moment(`2020-01-01T${free_time.free_to}`).format('LT');
    }

    if (parking_charges !== undefined) {
      if (parking_charges.length !== 0) {
        parking_charges.map((item, index) => {
          // add prefix to make it have a format that moment can handle
          start = moment(`2020-01-01T${item.time_start}`).format('LT');
          end = moment(`2020-01-01T${item.time_end}`).format('LT');
          list_time.push(start + ' - ' + end);
          list_price.push(formatMoney(item.price_per_hour) + '/h');
        });
      } else {
        list_time = 'Free';
      }
    }

    if (list_price.length !== 0) {
      time_now = list_time.shift();
      price_now = list_price.shift();
    }

    let timeHandler;
    if ((status === 'FULL' && !spot_name) || (status === 'ĐẦY' && !spot_name)) {
      titleColor = Colors.Red;
      timeHandler = setTimeout(async () => {
        const location = searchedLocation
          ? { lat: searchedLocation.latitude, lng: searchedLocation.longitude }
          : await getCurrentLatLng();
        const { data, success } = await axiosGet(API.PARKING.NEAREST, {
          params: { lat: location.lat, lng: location.lng },
        });
        if (success) {
          const res = { parking_nearest: data, status: 'parking_nearest' };
          navigate(Routes.MapDashboard, {
            scanDataResponse: res,
          });
        }
      }, 4000);
    }

    useEffect(() => {
      return () => clearTimeout(timeHandler);
    });

    if (status === 'FREE' || status === 'MIỄN PHÍ') {
      available_spots_count = '--';
    }

    return (
      <>
        <Text type="H4" semibold style={styles.title} color={Colors.Gray9}>
          {name}
        </Text>
        <Text type="Body" style={styles.address} color={Colors.Gray8}>
          {address}
        </Text>

        <ParkingStatusBar
          testID={TESTID.PARKING_DETAIL_STATUS_BAR}
          status={status}
          freeFrom={freeFrom}
          freeTo={freeTo}
          preBook={preBook}
          spot_name
        />

        <RowItem
          testID={TESTID.PARKING_DETAIL_SPOTS_AVAILABLE}
          source={<SvgParking width={16} height={16} />}
          title={`${available_spots_count}`}
          titleColor={titleColor}
          subTitle={`/ ${total_spot} ${t('spots_available')}`}
        />
        {price_now !== '' ? (
          <View>
            <ExpandView
              leftIcon={
                <IconOutline
                  name={'dollar-circle'}
                  size={16}
                  color={Colors.Gray8}
                />
              }
              title={price_now}
              expandedView={
                <>
                  {list_price.map((price) => (
                    <Text
                      key={price}
                      type={'H4'}
                      color={Colors.Gray9}
                      style={styles.textExpand}
                    >
                      {price}
                    </Text>
                  ))}
                </>
              }
            />
            <ExpandView
              leftIcon={
                <IconOutline
                  name={'clock-circle'}
                  size={16}
                  color={Colors.Gray8}
                />
              }
              title={time_now}
              expandedView={
                <>
                  {list_time.map((time) => (
                    <Text
                      key={time}
                      type={'H4'}
                      color={Colors.Gray9}
                      style={styles.textExpand}
                    >
                      {time}
                    </Text>
                  ))}
                  <Text
                    type={'Body'}
                    bold
                    color={Colors.Green}
                    style={styles.textExpand}
                  >
                    {t('text_no_charge_outside_parking_hour')}
                  </Text>
                </>
              }
            />
          </View>
        ) : (
          <View>
            <RowItem
              source={
                <IconOutline
                  name={'clock-circle'}
                  size={16}
                  color={Colors.Gray8}
                />
              }
              title="0 đ/h"
            />
            <RowItem
              source={
                <IconOutline
                  name={'clock-circle'}
                  size={16}
                  color={Colors.Gray8}
                />
              }
              title={list_time}
            />
          </View>
        )}

        <RowItem
          testID={TESTID.PARKING_DETAIL_DISTANCE}
          source={<SvgNavigate width={16} height={16} />}
          title={distance}
        />
        <View style={styles.line} />
      </>
    );
  }
);

export default AddressInfo;

const styles = StyleSheet.create({
  title: {
    marginTop: 16,
    marginLeft: 16,
    marginBottom: 8,
  },
  address: {
    marginBottom: 16,
    marginLeft: 16,
  },
  line: {
    height: 1,
    backgroundColor: Colors.Gray4,
    marginVertical: 16,
  },
  textExpand: {
    marginLeft: 36,
    marginTop: 8,
  },
});
