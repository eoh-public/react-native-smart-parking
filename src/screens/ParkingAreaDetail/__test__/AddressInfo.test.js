import React from 'react';
import { act, create } from 'react-test-renderer';
import AddressInfo from '../AddressInfo';

const mockGetCurrentLatLng = jest.fn();

jest.mock('utils/CountryUtils', () => {
  return {
    ...jest.requireActual('utils/CountryUtils'),
    getCurrentLatLng: mockGetCurrentLatLng,
  };
});

mockGetCurrentLatLng.mockReturnValue({
  lat: 10.762622,
  lng: 106.660172,
});

describe('Test AddressInfo', () => {
  let data;

  beforeEach(() => {
    data = {
      name: '',
      address: '',
      available_spots_count: '',
      parking_charges: [
        {
          time_start: '12:00:00.000Z',
          time_end: '12:00:00.000Z',
        },
      ],
      free_time: {
        free_from: '12:00:00.000Z',
        free_to: '12:00:00.000Z',
      },
      status: null,
      distance: 10000,
      total_spot: 1,
      preBook: false,
    };
  });
  let tree;

  test('create AddressInfo', () => {
    act(() => {
      tree = create(<AddressInfo {...data} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('create AddressInfo with free case', () => {
    data.parking_charges = [];
    act(() => {
      tree = create(<AddressInfo {...data} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('create AddressInfo with expand view list price/time', () => {
    data.parking_charges = [
      {
        time_start: '12:00:00.000Z',
        time_end: '12:00:00.000Z',
      },
      {
        time_start: '12:00:00.000Z',
        time_end: '12:00:00.000Z',
      },
    ];
    act(() => {
      tree = create(<AddressInfo {...data} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('create AddressInfo with status full', async () => {
    jest.useFakeTimers();
    data.status = 'FULL';
    act(() => {
      tree = create(<AddressInfo {...data} />);
    });
    act(() => {
      jest.runAllTimers();
    });

    // expect(mockGetCurrentLatLng).toHaveBeenCalled(); // TODO not working
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
