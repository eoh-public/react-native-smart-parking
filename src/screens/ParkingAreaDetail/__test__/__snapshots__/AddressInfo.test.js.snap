// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AddressInfo create AddressInfo 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#262626",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": 16,
          "lineHeight": 24,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginBottom": 8,
          "marginLeft": 16,
          "marginTop": 16,
        },
      ]
    }
  >
    
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 14,
          "lineHeight": 22,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginBottom": 16,
          "marginLeft": 16,
        },
      ]
    }
  >
    
  </Text>,
  <View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "column",
            "justifyContent": "center",
            "marginBottom": 16,
            "marginLeft": 16,
            "marginRight": 16,
          },
          Object {
            "backgroundColor": "#F5F5F5",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginLeft": 20,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#595959",
                "fontFamily": "antoutline",
                "fontSize": 20,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "paddingRight": 16,
              },
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": undefined,
                "lineHeight": undefined,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "color": "#8C8C8C",
              },
            ]
          }
        >
          Bãi đỗ xe không hỗ trợ đặt chỗ trước.
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingLeft": 16,
        "paddingVertical": 8,
      }
    }
  >
    <SvgMock
      height={16}
      width={16}
    />
    <Text
      style={
        Array [
          Object {
            "color": "#262626",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 16,
            "lineHeight": 24,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "marginLeft": 18,
            "marginTop": 0,
          },
        ]
      }
    >
      
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
         
        / 1 chỗ trống
         
      </Text>
    </Text>
  </View>,
  <View>
    <View
      style={
        Object {
          "marginVertical": 8,
          "paddingHorizontal": 16,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#595959",
                "fontFamily": "antoutline",
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              undefined,
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#262626",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 16,
                "lineHeight": 24,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "flex": 1,
                "marginLeft": 18,
                "marginTop": 0,
              },
            ]
          }
        >
          NaN đ/h
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "black",
                "fontFamily": "antoutline",
                "fontSize": 15,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              undefined,
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginVertical": 8,
          "paddingHorizontal": 16,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#595959",
                "fontFamily": "antoutline",
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              undefined,
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#262626",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 16,
                "lineHeight": 24,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "flex": 1,
                "marginLeft": 18,
                "marginTop": 0,
              },
            ]
          }
        >
          12:00 - 12:00
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "black",
                "fontFamily": "antoutline",
                "fontSize": 15,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              undefined,
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingLeft": 16,
        "paddingVertical": 8,
      }
    }
  >
    <SvgMock
      height={16}
      width={16}
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 16,
            "lineHeight": 24,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "marginLeft": 18,
            "marginTop": 0,
          },
        ]
      }
    >
      10km
    </Text>
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#E8E8E8",
        "height": 1,
        "marginVertical": 16,
      }
    }
  />,
]
`;

exports[`Test AddressInfo create AddressInfo with expand view list price/time 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#262626",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": 16,
          "lineHeight": 24,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginBottom": 8,
          "marginLeft": 16,
          "marginTop": 16,
        },
      ]
    }
  >
    
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 14,
          "lineHeight": 22,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginBottom": 16,
          "marginLeft": 16,
        },
      ]
    }
  >
    
  </Text>,
  <View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "column",
            "justifyContent": "center",
            "marginBottom": 16,
            "marginLeft": 16,
            "marginRight": 16,
          },
          Object {
            "backgroundColor": "#F5F5F5",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginLeft": 20,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#595959",
                "fontFamily": "antoutline",
                "fontSize": 20,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "paddingRight": 16,
              },
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": undefined,
                "lineHeight": undefined,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "color": "#8C8C8C",
              },
            ]
          }
        >
          Bãi đỗ xe không hỗ trợ đặt chỗ trước.
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingLeft": 16,
        "paddingVertical": 8,
      }
    }
  >
    <SvgMock
      height={16}
      width={16}
    />
    <Text
      style={
        Array [
          Object {
            "color": "#262626",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 16,
            "lineHeight": 24,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "marginLeft": 18,
            "marginTop": 0,
          },
        ]
      }
    >
      
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
         
        / 1 chỗ trống
         
      </Text>
    </Text>
  </View>,
  <View>
    <View
      style={
        Object {
          "marginVertical": 8,
          "paddingHorizontal": 16,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#595959",
                "fontFamily": "antoutline",
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              undefined,
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#262626",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 16,
                "lineHeight": 24,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "flex": 1,
                "marginLeft": 18,
                "marginTop": 0,
              },
            ]
          }
        >
          NaN đ/h
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "black",
                "fontFamily": "antoutline",
                "fontSize": 15,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              undefined,
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginVertical": 8,
          "paddingHorizontal": 16,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#595959",
                "fontFamily": "antoutline",
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              undefined,
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#262626",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 16,
                "lineHeight": 24,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "flex": 1,
                "marginLeft": 18,
                "marginTop": 0,
              },
            ]
          }
        >
          12:00 - 12:00
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "black",
                "fontFamily": "antoutline",
                "fontSize": 15,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              undefined,
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingLeft": 16,
        "paddingVertical": 8,
      }
    }
  >
    <SvgMock
      height={16}
      width={16}
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 16,
            "lineHeight": 24,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "marginLeft": 18,
            "marginTop": 0,
          },
        ]
      }
    >
      10km
    </Text>
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#E8E8E8",
        "height": 1,
        "marginVertical": 16,
      }
    }
  />,
]
`;

exports[`Test AddressInfo create AddressInfo with free case 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#262626",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": 16,
          "lineHeight": 24,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginBottom": 8,
          "marginLeft": 16,
          "marginTop": 16,
        },
      ]
    }
  >
    
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 14,
          "lineHeight": 22,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginBottom": 16,
          "marginLeft": 16,
        },
      ]
    }
  >
    
  </Text>,
  <View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "column",
            "justifyContent": "center",
            "marginBottom": 16,
            "marginLeft": 16,
            "marginRight": 16,
          },
          Object {
            "backgroundColor": "#F5F5F5",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginLeft": 20,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#595959",
                "fontFamily": "antoutline",
                "fontSize": 20,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "paddingRight": 16,
              },
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": undefined,
                "lineHeight": undefined,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "color": "#8C8C8C",
              },
            ]
          }
        >
          Bãi đỗ xe không hỗ trợ đặt chỗ trước.
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingLeft": 16,
        "paddingVertical": 8,
      }
    }
  >
    <SvgMock
      height={16}
      width={16}
    />
    <Text
      style={
        Array [
          Object {
            "color": "#262626",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 16,
            "lineHeight": 24,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "marginLeft": 18,
            "marginTop": 0,
          },
        ]
      }
    >
      
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
         
        / 1 chỗ trống
         
      </Text>
    </Text>
  </View>,
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingLeft": 16,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#595959",
              "fontFamily": "antoutline",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 16,
              "lineHeight": 24,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "marginLeft": 18,
              "marginTop": 0,
            },
          ]
        }
      >
        0 đ/h
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingLeft": 16,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#595959",
              "fontFamily": "antoutline",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 16,
              "lineHeight": 24,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "marginLeft": 18,
              "marginTop": 0,
            },
          ]
        }
      >
        Free
      </Text>
    </View>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingLeft": 16,
        "paddingVertical": 8,
      }
    }
  >
    <SvgMock
      height={16}
      width={16}
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 16,
            "lineHeight": 24,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "marginLeft": 18,
            "marginTop": 0,
          },
        ]
      }
    >
      10km
    </Text>
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#E8E8E8",
        "height": 1,
        "marginVertical": 16,
      }
    }
  />,
]
`;

exports[`Test AddressInfo create AddressInfo with status full 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#262626",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": 16,
          "lineHeight": 24,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginBottom": 8,
          "marginLeft": 16,
          "marginTop": 16,
        },
      ]
    }
  >
    
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 14,
          "lineHeight": 22,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginBottom": 16,
          "marginLeft": 16,
        },
      ]
    }
  >
    
  </Text>,
  <View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "column",
            "justifyContent": "center",
            "marginBottom": 16,
            "marginLeft": 16,
            "marginRight": 16,
          },
          Object {
            "backgroundColor": "#F5F5F5",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginLeft": 20,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#595959",
                "fontFamily": "antoutline",
                "fontSize": 20,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "paddingRight": 16,
              },
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": undefined,
                "lineHeight": undefined,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "color": "#8C8C8C",
              },
            ]
          }
        >
          Bãi đỗ xe không hỗ trợ đặt chỗ trước.
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingLeft": 16,
        "paddingVertical": 8,
      }
    }
  >
    <SvgMock
      height={16}
      width={16}
    />
    <Text
      style={
        Array [
          Object {
            "color": "red",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 16,
            "lineHeight": 24,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "marginLeft": 18,
            "marginTop": 0,
          },
        ]
      }
    >
      
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
         
        / 1 chỗ trống
         
      </Text>
    </Text>
  </View>,
  <View>
    <View
      style={
        Object {
          "marginVertical": 8,
          "paddingHorizontal": 16,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#595959",
                "fontFamily": "antoutline",
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              undefined,
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#262626",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 16,
                "lineHeight": 24,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "flex": 1,
                "marginLeft": 18,
                "marginTop": 0,
              },
            ]
          }
        >
          NaN đ/h
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "black",
                "fontFamily": "antoutline",
                "fontSize": 15,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              undefined,
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginVertical": 8,
          "paddingHorizontal": 16,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#595959",
                "fontFamily": "antoutline",
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              undefined,
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#262626",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 16,
                "lineHeight": 24,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "flex": 1,
                "marginLeft": 18,
                "marginTop": 0,
              },
            ]
          }
        >
          12:00 - 12:00
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "black",
                "fontFamily": "antoutline",
                "fontSize": 15,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              undefined,
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingLeft": 16,
        "paddingVertical": 8,
      }
    }
  >
    <SvgMock
      height={16}
      width={16}
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 16,
            "lineHeight": 24,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "marginLeft": 18,
            "marginTop": 0,
          },
        ]
      }
    >
      10km
    </Text>
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#E8E8E8",
        "height": 1,
        "marginVertical": 16,
      }
    }
  />,
]
`;
